pkg/services/ngalert/state/cache.go

```diff
@@ -210,7 +200,6 @@ func calculateState(ctx context.Context, log log.Logger, alertRule *ngModels.Ale
 		Labels:             lbs,
 		Annotations:        annotations,
 		EvaluationDuration: result.EvaluationDuration,
-		Values:             values,
```

Comment:
This is the core of the fix - moving the `Values` logic over to `setNextState`.

This has zero effect on templates. Template expansion already happened before this change. There's no reason for this to be in this function

Reply:
I think it makes sense to remove this assignment too.
https://github.com/grafana/grafana/blob/47626c5d989febf2207401fdd558a4fb610f5343/pkg/services/ngalert/state/cache.go#L117